app/controllers/application_controller.rb

```diff
@@ -16,4 +16,23 @@ def configure_permitted_parameters
                profile_attributes: %i[id name city country description])
     end
   end
+
+  def find_friends
```

Comment:
This method isn't the responsibility of the `ApplicationController`, it should be handled by the User or Friendship classes. I understand you need it in the PostController too but you need to find a different way to use it without sacrificing the Single Responsibility Principle.